Практика ООП

Revanit opened the pull request:
Добавил файлы Figure, Triangle, Circle, Rectangle, Square
Добавил тесты test_Triangle, test_Circle, test_Rectangle, test_Square,

VDA-VI committed:
Добавил файлы Figure, Triangle, Circle, Rectangle, Square

Добавил тесты test_Triangle, test_Circle, test_Rectangle, test_Square,

e7506c commented on tests/test_circle.py:
```diff
@@ -0,0 +1,47 @@
+
+
+def test_circle_add_area():
+    c1 = Circle(1)
```
тестовые данные лучше вынести в параметризацию, а не хардкодить в тесте:

```
@pytest.mark.parametrize(
    "figure_1, figure_2",
    [
        (Circle(1), Circle(2))
    ]
)
```

e7506c commented on tests/test_rectangle.py:
```diff
@@ -1,26 +1,46 @@
+
+
+def test_rectangle_add_area():
+    r = Rectangle(3, 4)
```
лучше вынести в параметризацию

VDA-VI committed:
Добавил файлы Script и Result

VDA-VI committed:
Добавил тесты следующих сервисов:

https://dog.ceo/dog-api/
https://www.openbrewerydb.org/
https://jsonplaceholder.typicode.com/

Добавил Тестовую функцию с pytest.addoption

e7506c commented on tests/test_brewery_api.py:
```diff
@@ -0,0 +1,37 @@
+    def test_get_breweries_by_city(self, by_city):
+        response = requests.get(f"{BASE_URL}/breweries", params={"by_city": by_city})
+        assert response.status_code == 200
+        assert isinstance(response.json(), list)
```
а где проверка, что вернулись данные для нужного города?

e7506c commented on tests/test_brewery_api.py:
```diff
@@ -0,0 +1,37 @@
+
+    def test_get_single_brewery(self):
+        response = requests.get(
+            f"{BASE_URL}/breweries/b54b16e1-ac3b-4bff-a11f-f7ae9ddc27e0"
```
не нужно хардкодить значения в тесте. лучше вынести их в параметризацию

e7506c commented on tests/test_brewery_api.py:
```diff
@@ -0,0 +1,37 @@
+        assert data["id"] == "b54b16e1-ac3b-4bff-a11f-f7ae9ddc27e0"
+
+    def test_search_brewery(self):
+        response = requests.get(f"{BASE_URL}/breweries/search", params={"query": "dog"})
```
аналогично - вынести данные в параметризацию

e7506c commented on tests/test_brewery_api.py:
```diff
@@ -0,0 +1,37 @@
+    def test_search_brewery(self):
+        response = requests.get(f"{BASE_URL}/breweries/search", params={"query": "dog"})
+        assert response.status_code == 200
+        assert isinstance(response.json(), list)
```
а где проверка, что данные, которые вернул сервис, как-то зависят от нашего запроса?

e7506c commented on tests/test_dog_api.py:
```diff
@@ -0,0 +1,41 @@
+        assert response.status_code == 200
+        data = response.json()
+        assert data["status"] == "success"
+        assert isinstance(data["message"], list)
```
нужна проверка, что вернулись данные для нужной породы.

e7506c commented on tests/test_dog_api.py:
```diff
@@ -0,0 +1,41 @@
+        response = requests.get(f"{BASE_URL}/breed/hound/{sub_breed}/images")
+        assert response.status_code == 200
+        data = response.json()
+        assert data["status"] == "success"
```
проверка, что вернулись нужные данные

e7506c commented on tests/test_dog_api.py:
```diff
@@ -0,0 +1,41 @@
+        response = requests.get(f"{BASE_URL}/breed/hound/images/random")
+        assert response.status_code == 200
+        data = response.json()
+        assert data["status"] == "success"
```
в чем отличие от предыдущего теста, если вы не проверяете, что вернулось что-то для hound ?

VDA-VI committed:
Внёс правки после ревью

VDA-VI committed:
фикс flake8
